lib/fpm/package/cpan.rb

```diff
@@ -251,6 +240,27 @@ def input(package)
     end
   end
 
+  def export_local_lib_env(local_lib_path)
+    # Iterate over the output of local::lib line by line
+    `#{attributes[:cpan_perl_bin]} -Mlocal::lib=#{local_lib_path}`.lines.each do |line|
+      # local::lib's output is in the form:
+      # 'ENV_VAR="some value"; export $ENV_VAR;'
+      # This regular expression strips everything from the semicolon in front
+      # of 'export' to the end of the line, then splits on the equals sign,
+      # limiting the returned array to two elements.
+      (k, v) = line.chomp.gsub(/;\s+export.*$/, "").split("=", 2)
+
+      # Skip these, since they wreak havoc when PREFIX # is set.
+      next if k == "PERL_MM_OPT" or k == "PERL_MB_OPT"
+
+      # Set the current environment in accordance with local::lib's output.
+      # The subshell call to echo is necessary because some environment
+      # variables are defined in terms of other such variables; this technique
+      # expands the variables-within-variables.
+      ENV[k] = `echo #{v}`.chomp
```

Comment:
Thank you for the comment explaining why you do this subshell echo. I was super confused until I read the comment :heart: